target/riscv/cpu.h

```diff
@@ -586,4 +591,29 @@ riscv_cpu_get_tb_cpu_state(CPURISCVState *env, target_ulong *pc,
 // Ugly macro hack to avoid having to modify cpu_get_tb_cpu_state in all targets
 #define cpu_get_tb_cpu_state riscv_cpu_get_tb_cpu_state
 
+#ifdef CONFIG_TCG_LOG_INSTR
+/*
+ * Target-specific hook to fetch the cpu log state
+ */
+static inline cpu_log_instr_info_t *cpu_get_log_instr_state(CPURISCVState *env)
+{
+    return NULL;
+}
+
+/* TODO(am2419): Document these as required to support a new target.
+ * New common log API arch-specific helpers.
+ */
+static inline bool cpu_in_user_mode(CPURISCVState *env)
+{
+    return env->priv == PRV_U;
+}
+
+// TODO(am2419): deprecate and remove 
+// TODO(am2419) should probably rename as cpu_get_asid()
```

Comment:
That TODO has been addressed.

I'm not sure we really need this function, but it was useful in the past to filter logs.